provider/confluence/api.ts

```diff
@@ -0,0 +1,123 @@
+import type { Settings } from './index.ts'
+
+const authHeaders = (settings: Settings) => ({
+    Authorization: `Basic ${Buffer.from(`${settings.email}:${settings.apiToken}`).toString('base64')}`,
+})
+
+const buildUrl = (settings: Settings, path: string, searchParams: Record<string, string> = {}) => {
+    const url = new URL(settings.url.replace(/\/$/, '') + path)
+    url.search = new URLSearchParams(searchParams).toString()
+    return url
+}
+
+export interface PageSearchResult {
+    id: string
+    title: string
+    space: {
+        name: string
+    }
+    uri: string
+}
+
+interface PagesSearchJSON {
+    results: {
+        id: string
+        title: string
+        space: {
+            name: string
+        }
+        _links: {
+            webui: string
+        }
+    }[]
+}
+
+export interface Page {
+    id: string
+    title: string
+    space: {
+        name: string
+    }
+    body: string
+    uri: string
+}
+
+interface PageJSON {
+    id: string
+    title: string
+    space: {
+        name: string
+    }
+    body: {
+        storage: {
+            value: string
+        }
+    }
+    _links: {
+        webui: string
+    }
+}
+
+export const listPages = async (settings: Settings, query?: string): Promise<PageSearchResult[]> => {
+    const url = buildUrl(settings, '/rest/api/content/search', {
+        cql: `type=page AND ${
+            query ? `title ~ ${JSON.stringify(query + '*')}` : 'title ~ "*"'
```

Comment:
does it need `*` on both sides of query?

Reply:
You'd think so, but CQL which uses Lucene doesn't support that. If you do that, it will match nothing 😅 https://developer.atlassian.com/cloud/confluence/performing-text-searches-using-cql/#wildcard-searches----and--

JIRA on the other hand has a "picker" API which is designed for this type of fuzzy/autocomplete use case: https://docs.atlassian.com/software/jira/docs/api/REST/9.16.0/#api/2/issue-getIssuePickerResource

I couldn't find an equivalent API for Confluence.